github-connector/internal/events/sender_test.go

```diff
@@ -0,0 +1,168 @@
+package events_test
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/kyma-incubator/hack-showcase/github-connector/internal/apperrors"
+	"github.com/kyma-incubator/hack-showcase/github-connector/internal/events"
+	"github.com/kyma-incubator/hack-showcase/github-connector/internal/events/mocks"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type toJSON struct {
+	TestJSON string `json:TestJSON"`
+}
+
+type ClientMock struct {
+}
+
+func (c *ClientMock) Do(req *http.Request) (*http.Response, error) {
+	return &http.Response{}, nil
+}
+
+func TestSendToKyma(t *testing.T) {
+	t.Run("should return no error when given proper arguments", func(t *testing.T) {
+		k := events.NewSender(&ClientMock{}, events.NewValidator(), "http://event-bus-publish.kyma-system:8080/v1/events")
+		payload := toJSON{TestJSON: "test"}
+		toSend, err := json.Marshal(payload)
+		require.NoError(t, err)
+		assert.Equal(t, nil, k.SendToKyma("issuesevent.opened", "v1", "", "github-connector-app", json.RawMessage(toSend)))
+	})
+
+	t.Run("should return an internal error when given no event type", func(t *testing.T) {
```

Comment:
The following tests:
- "should return an internal error when given no event type"
- "should return an internal error when given no event version"
- "should return an internal error when given no source id"

Exercise exactly the same scenario: namely the case when validator returns some error. Since we mock Validator dependency we don't need to cover all the possible cases when validation fails.